app/controllers/foods_controller.rb

```diff
@@ -0,0 +1,47 @@
+class FoodsController < ApplicationController
+  before_action :set_food, only: %i[destroy]
+
+  # GET /foods or /foods.json
+  def index
+    @foods = current_user.foods
+  end
+
+  # GET /foods/new
+  def new
+    @food = current_user.foods.build
+  end
+
+  # POST /foods or /foods.json
+  def create
+    @food = Food.new(food_params)
+
+    respond_to do |format|
+      if @food.save
+        format.html { redirect_to '/foods', notice: 'Food was successfully created.' }
+      else
+        format.html { render :new, status: :unprocessable_entity }
+      end
+    end
+  end
+
+  # DELETE /foods/1 or /foods/1.json
+  def destroy
+    @food.destroy!
+
+    respond_to do |format|
```

Comment:
- [ ] Let's ensure to optimize our database queries by introducing solutions to **N + 1** problem in the code so as to improve the app's performance. For instance, instead of fetching all books here, the last code will fetch books and the associated authors in a single instead making two different queries. This improves the performance.

```ruby
class BooksController < ApplicationController
  def index
    @books = Book.all
  end
end
```

We can do:
```ruby
class BooksController < ApplicationController
  def index
    @books = Book.includes(:author).all
  end
end
```